services/src/main/java/org/keycloak/protocol/oidc/AccessTokenIntrospectionProvider.java

```diff
@@ -58,13 +61,18 @@ public Response introspect(String token) {
 
                 PublicKey publicKey = session.keys().getRsaPublicKey(realm, verifier.getHeader().getKeyId());
                 if (publicKey == null) {
+                    logger.debug("Verification failed: public key not found");
                     valid = false;
                 } else {
                     verifier.publicKey(publicKey);
                     verifier.verify();
                     toIntrospect = verifier.getToken();
                 }
             } catch (VerificationException e) {
+                logger.debug("Verification failed: " + e.getMessage());
```

Comment:
The reason I didn't was to avoid a massive stack trace that slf4j would produce, the message from the verification exception message concisely contains what I think is needed.
Happy to change it if you think the stack trace is useful.